perl/lib/BarnOwl/Message/Zephyr.pm

```diff
@@ -222,5 +241,65 @@ sub replysendercmd {
     return $self->replycmd(1);
 }
 
+# Logging
+sub log_header {
+    my ($m) = @_;
+    my $class = $m->class;
+    my $instance = $m->instance;
+    my $opcode = $m->opcode;
+    my $timestr = $m->time;
+    my $host = $m->host;
+    my $sender = $m->pretty_sender;
+    my $zsig = $m->zsig;
+    my $rtn = "Class: $class Instance: $instance";
+    $rtn .= " Opcode: $opcode" unless !defined $opcode || $opcode eq '';
+    $rtn .= "\nTime: $timestr Host: $host"
+          . "\nFrom: $zsig <$sender>";
+    return $rtn;
```

Comment:
I got about this far without serious issues. I'm now tired enough that I'm not convinced further review will be reliable to leave me confident saying "LGTM", so I'm stopping for the night.